bundles/org.eclipse.swt/Eclipse SWT/cocoa/org/eclipse/swt/widgets/Display.java

```diff
@@ -117,6 +117,18 @@ enum APPEARANCE {
 	/* System property to be set for SWT application to use the system's theme */
 	static final String USE_SYSTEM_THEME = "org.eclipse.swt.display.useSystemTheme";
 
+	/**
+	 * This constant is used for configuring whether control.forceFocus() activates the parent shell.
+	 * Expects a <code>boolean</code> value.
+	 * To prevent control.forceFocus() to activate the shell, invoke
+	 * <pre>
+	 *   display.setData(Display.ACTIVATE_SHELL_ON_FORCE_FOCUS, false);
+	 * </pre>
+	 * after having created the display.
+	 */
+	public static final String ACTIVATE_SHELL_ON_FORCE_FOCUS = "org.eclipse.swt.activateShellOnForceFocus"; //$NON-NLS-1$
```

Comment:
This adds **public** API and API tools reports an error (missing since 3.123).
**If** the field need to be public, it needs a proper \@since comment.

Reply:
IMHO it makes sense to have this constant being public, so it could be invoked in the user's code which usually uses different packages. In the new pull request I've added the @since tag, but I'm unsure about the version - I've used `@since 4.959` because this is the version that occurs in the `version.txt` inside the swt.jar.

Reply:
Thomas, you should use the version defined in the manifest.mf. This is what is checked by API tooling to make sure the API's aren't broken / changed on next release without proper version update.